tests/Integration/RequestIntegrationTest.php

```diff
@@ -8,52 +8,77 @@
 use Psr\Http\Message\ServerRequestInterface;
 use Sentry\Event;
 use Sentry\Integration\RequestIntegration;
+use Sentry\Options;
 use Zend\Diactoros\ServerRequest;
 use Zend\Diactoros\Uri;
 
 final class RequestIntegrationTest extends TestCase
 {
     /**
-     * @dataProvider invokeDataProvider
+     * @dataProvider invokeUserContextPiiDataProvider
      */
-    public function testInvoke(array $requestData, array $expectedValue): void
+    public function testInvokeWithRequestHavingIpAddress(bool $pii, array $expectedValue): void
     {
         $event = new Event();
+        $event->getUserContext()->setData(['foo' => 'bar']);
 
         $request = new ServerRequest();
-        $request = $request->withCookieParams($requestData['cookies']);
-        $request = $request->withUri(new Uri($requestData['uri']));
-        $request = $request->withMethod($requestData['method']);
-
-        foreach ($requestData['headers'] as $name => $value) {
-            $request = $request->withHeader($name, $value);
-        }
+        $request = $request->withHeader('REMOTE_ADDR', '127.0.0.1');
 
         $this->assertInstanceOf(ServerRequestInterface::class, $request);
 
-        RequestIntegration::applyToEvent($event, $request);
+        $integration = new RequestIntegration(new Options(['send_default_pii' => $pii]));
+
+        RequestIntegration::applyToEvent($integration, $event, $request);
 
-        $this->assertEquals($expectedValue, $event->getRequest());
+        $this->assertEquals($expectedValue, $event->getUserContext()->toArray());
     }
 
-    public function testInvokeWithRequestHavingIpAddress(): void
+    public function invokeUserContextPiiDataProvider(): array
+    {
+        return [
+            [
+                true,
+                ['ip_address' => '127.0.0.1', 'foo' => 'bar'],
+            ],
+            [
+                false,
+                ['foo' => 'bar'],
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider invokeDataProvider
+     */
+    public function testInvoke(bool $pii, array $requestData, array $expectedResult): void
```

Comment:
Can we rename the variable `$pii` to something that better describes its purpose (e.g. `$shouldSendPii`)?